test/serialization.test.ts

```diff
@@ -0,0 +1,116 @@
+import { SRPParameters } from "../src/parameters";
+import { SRPRoutines } from "../src/routines";
+import {
+  SRPClientSession,
+  SRPClientSessionStep1,
+  SRPClientSessionStep2,
+} from "../src/session-client";
+import { SRPServerSession, SRPServerSessionStep1 } from "../src/session-server";
+import { createVerifierAndSalt, generateRandomString } from "../src/utils";
+import { serialize, deserialize } from "../src/serde";
+import { test } from "./tests";
+
+const TEST_ROUTINES = new SRPRoutines(new SRPParameters());
+
+test("#Server serde", async (t) => {
+  const testUsername = generateRandomString(10);
+  const testPassword = generateRandomString(15);
+
+  const { s: salt, v: verifier } = await createVerifierAndSalt(
+    TEST_ROUTINES,
+    testUsername,
+    testPassword,
+  );
+
+  // serialized server step1, and B
+  // B is sent back to client, server_str is stored into some kind of webserver
+  // session for this user
+  const [server_str, B] = await (async (): Promise<[string, bigint]> => {
+    const server = await new SRPServerSession(TEST_ROUTINES).step1(
+      testUsername,
+      salt,
+      verifier,
+    );
+    const B = server.B;
+
+    return [serialize(server), B];
+  })();
+
+  const client = await new SRPClientSession(TEST_ROUTINES).step1(
+    testUsername,
+    testPassword,
+  );
+
+  const client_step2 = await client.step2(salt, B);
+
+  // assume that server_str was stored/loaded from some session
+  const server_step1 = deserialize(server_str, SRPServerSessionStep1.prototype);
```

Comment:
I don't think SRPServerSessionStep1 and other step classes are exported. If we go with this approach they need to be.

Reply:
https://github.com/midonet/tssrp6a/blob/1303f8d23754015f3cf26ab9690ca093f9b4ade2/src/index.ts#L3-L4

Reply:
Yes, you are right! My bad